### PR TITLE
chore: Move to VS2022

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Details

This PR moves us from VS2019 to VS2022. The solution, project and resource files were manually updated. The pipeline files were manually edited and required some trial and error. I originally updated the Roslyn analyzers version number from 16.0 to 17.0, but that failed, so they're intentionally remaining at 16.0. In the test tasks, I tried both 16.0 and 17.0 for `vsTestVersion`. Reading the [docs](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/test/vstest?view=azure-devops), though, this parameter is optional and things worked once I removed the parameter.

A successful run of the signed pipeline can be found [here](https://dev.azure.com/mseng/1ES/_build/results?buildId=17666697&view=results).

##### Motivation

Use the current tools

#### Pull request checklist

- [n/a] Addresses an existing issue

### CODE REVIEW

File: build/signedbuild.yml [hunk 1]
```diff
@@ -134,7 +133,7 @@ jobs:
         userProvideBuildInfo: msBuildInfo
         msbuildVersion: 16.0
         msBuildArchitecture: '$(BuildPlatform)'
-        msBuildCommandline: '"C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Current\Bin\msbuild.exe" "$(Build.SourcesDirectory)\src\AccessibilityInsights.sln" /p:platform="$(BuildPlatform)" /p:configuration="debug" /p:VisualStudioVersion="16.0"'
+        msBuildCommandline: '"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\msbuild.exe" "$(Build.SourcesDirectory)\src\AccessibilityInsights.sln" /p:platform="$(BuildPlatform)" /p:configuration="debug" /p:VisualStudioVersion="16.0"'
```
Comment (DaveTryon) [inline build/signedbuild.yml:136]: It's unfortunate to have the hardcoded path here, but there's no great environment variable we can leverage. On the optimistic side, we've been using a hardcoded path for about 3 years without problem, so it seems unlikely to be an issue until we need to change files to support Visual Studio 18

File: src/AccessibilityInsights.CustomActions/CustomActions.csproj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (DaveTryon) [inline src/AccessibilityInsights.CustomActions/CustomActions.csproj:1]: VS inserted a UTF-8 BOM at the head of this file

**Review by jalkire — APPROVED** [review]

File: build/signedbuild.yml (hunk 1 above)
Comment (jalkire) [inline build/signedbuild.yml:136]:
```suggestion
        msBuildCommandline: '"C:\Program Files\Microsoft Visual Studio\2022\Enterprise\MSBuild\Current\Bin\msbuild.exe" "$(Build.SourcesDirectory)\src\AccessibilityInsights.sln" /p:platform="$(BuildPlatform)" /p:configuration="debug" /p:VisualStudioVersion="17.0"'
```

File: build/signedbuild.yml [hunk 3]
```diff
@@ -134,7 +133,7 @@ jobs:
         userProvideBuildInfo: msBuildInfo
         msbuildVersion: 16.0
```
Comment (jalkire) [inline build/signedbuild.yml:134]:
```suggestion
        msbuildVersion: 17.0
```

File: build/signedbuild.yml (hunk 1 above)
Comment (DaveTryon) [reply build/signedbuild.yml:136]: I tried this in [this build](https://dev.azure.com/mseng/1ES/_build/results?buildId=17711063&view=logs&j=efe68385-d402-5492-a2cf-bc47ec32e784&t=e38751e8-0dc8-5e3c-8d46-e18c39d37211&l=1753) of Axe.Windows, but it doesn't work 😦

File: build/build-and-unit-tests.yml [hunk 4]
```diff
@@ -58,7 +58,6 @@ jobs:
       testAssemblyVer2: |
```
[the remark this answers is not in the document]
Comment (DaveTryon) [reply build/build-and-unit-tests.yml:58]: Agreed

**Review by jalkire — APPROVED** [review]

**Pull request merged**